### ISSUE TITLE
Kuby can sometimes be attempting to push to old clusters

### ISSUE THREAD

**Issue by lazyatom** [getkuby/kuby-digitalocean#2]
I've noticed this issue while testing Kuby, where I'm frequently creating and destroying clusters -- I believe [the implementation here](https://github.com/getkuby/kuby-digitalocean/blob/ad69edb0ea9c23d8a3ddc966f8919987113b10c8/lib/kuby/digitalocean/provider.rb#L80-L92) means that kuby tries to use the cached kubeconfig.yaml file, even though the details provided to Kuby's configuration have changed, resulting in the error message `Error: Kubernetes cluster unreachable` when trying to run `kuby setup`.

I'd suggest we create a hash of the credentials, and include that into `kubeconfig_path` somehow, such that _any_ change to the credentials results in an immediate refresh of the Kubernetes configuration.

**Comment by camertron** [comment]
@lazyatom you're absolutely right, it just looks to see if the kubeconfig file exists, which won't pick up on changes made in your kuby config. I'll file an issue for this.

**Comment by camertron** [comment]
Here we go: https://github.com/getkuby/kuby-core/issues/40
